src/main/java/fi/aalto/cs/apluscourses/model/NoSuchComponentException.java

```diff
@@ -3,8 +3,8 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-public class NoSuchModuleException extends Exception {
-  public NoSuchModuleException(@NotNull String moduleName, @Nullable Throwable cause) {
+public class NoSuchComponentException extends Exception {
+  public NoSuchComponentException(@NotNull String moduleName, @Nullable Throwable cause) {
     super("Module '" + moduleName + " was not found'.", cause);
```

Comment:
Could change the parameter to `componentName` and the message to `"Component '" + copmonentName + ...`